### PR TITLE
.travis.yml: Move unsupported check to a script

### PR DESCRIPTION
**Pull request by jayvdb**
The unsupported check was previously a hack inside the main
Travis CI `script` block, requiring many workarounds
throughout the .travis.yml.
Travis CI now supports custom `script` blocks for each build
matrix entry, which allows the logic to be moved to an
external and more clear script.
In addition, all of the installation steps have been
removed, making these jobs much faster to complete.

Closes https://github.com/coala/coala-bears/issues/2182

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: .ci/check_unsupported.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# Non-zero exit code is what we want to check
+set +e
+
+# Enable capturing the non-zero exit status of coverage instead of tee
+set -o pipefail
+
+sed -i.bak -E '/^(mypy-lang|guess-language-spirit)/d' bear-requirements.txt
+
+coverage run setup.py install | tee setup.log
```
Comment (SanketDG) [inline .ci/check_unsupported.sh]: `coverage run`?

File: .ci/check_unsupported.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+# Non-zero exit code is what we want to check
+set +e
+
+# Enable capturing the non-zero exit status of coverage instead of tee
+set -o pipefail
+
+sed -i.bak -E '/^(mypy-lang|guess-language-spirit)/d' bear-requirements.txt
```
Comment (SanketDG) [inline .ci/check_unsupported.sh]: can you add comment on why this line was added

File: .ci/check_unsupported.sh (hunk 1 above)
Comment (jayvdb) [reply .ci/check_unsupported.sh]: Used to see what code was touched by the py27 & py34 code paths.

File: .ci/check_unsupported.sh (hunk 2 above)
Comment (jayvdb) [reply .ci/check_unsupported.sh]: sorry I forgot to copy that comment from `.travis.yml`
now that this unsupported code oath is not using pip to install dependencies, `guess-language-spirit` also doesnt install , as setuptools can only find the latter version of the package, and then it complains that the version isnt the same as we require.

File: .ci/check_unsupported.sh (hunk 1 above)
Comment (jayvdb) [reply .ci/check_unsupported.sh]: I've done a build removing this.  it should be ok.

File: .ci/check_unsupported.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+# Non-zero exit code is what we want to check
+set +e
+
+# Enable capturing the non-zero exit status of coverage instead of tee
```
Comment (Mixih) [inline .ci/check_unsupported.sh]: not coverage anymore.

File: .ci/check_unsupported.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+# Non-zero exit code is what we want to check
+set +e
+
```
Comment (SanketDG) [inline .ci/check_unsupported.sh]: can we have `set -x` for debug output, its always helpful

File: .ci/check_unsupported.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+# Enable capturing the non-zero exit status of coverage instead of tee
+set -o pipefail
+
+# mypy-lang and guess-language-spirit do not install on unsupported versions
+sed -i.bak -E '/^(mypy-lang|guess-language-spirit)/d' bear-requirements.txt
```
Comment (SanketDG) [inline .ci/check_unsupported.sh]: does the backup help at all?

File: .ci/check_unsupported.sh (hunk 4 above)
Comment (Mixih) [reply .ci/check_unsupported.sh]: isn't this going to make a mess though?

**Comment by jayvdb** [comment]
> is travis-ci.org/coala/coala-bears/jobs/318432474#L524 a problem?

Copying here...

```
$ .ci/check_unsupported.sh
ERROR:root:Module coalib is not found. Cannot do version checking.
```

No, this is the bears logic caused by importing `bears` in `setup.py` here https://github.com/coala/coala-bears/blob/master/setup.py#L8 .

Then it tries to install coala and fails.

In my next trick, that line disappears too:

https://github.com/coala/coala-bears/pull/2134/files#diff-2eeaed663bd0d25b7e608891384b7298L8

File: .ci/check_unsupported.sh (hunk 4 above)
Comment (jayvdb) [reply .ci/check_unsupported.sh]: done below

**Comment by Mixih** [comment]
no cache build: https://travis-ci.org/coala/coala-bears/builds/319120699?utm_source=github_status&utm_medium=notification

**Comment by Mixih** [comment]
ack 5e7ab42 abf9e57

**Pull request merged**